tests/typescript/conformance/interfaces/interfaceDeclarations/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -44,8 +51,17 @@ interface Derived extends Base, Base2 {
   x: { b: string };
 }
 
-interface Derived2 extends Base, Base2 { // error
-  x: { a: number, b: string };
+// comment above TSInterfaceDeclaration
+interface Derived2 extends Base, Base2 // comment above TSInterfaceDeclaration
+{
+  // comment next TSInterfaceDeclaration
+  // comment above TSTypeLiteral
+  // comment inside of ObjectTypeAnnotation
+  // comment below TSTypeLiteral
```

Comment:
@vjeux 

Mhh, I find it weird that for iterfaces all comments are always attached to the `TSInterfaceDeclaration`, whereas for classes and objects the comments are attached to the following node:

```ts
interface I {
  a: string;
  // this comment is attached to the `TSInterfaceDeclaration` node which is right under `Program`
  b: string;
}
```

```ts
class C {
  public a: string;
  // this comment is attached to the following `ClassProperty` node
  public b: string;
}
```

```ts
const c = {
  a: 1,
  // this comment attached to the following `Property` node
  b: 2,
}
```

To me it seems as if the printer decides where to put the comments. And for that is uses mostly the logic to print the comment before the node it is attached to. Seems in the case of `TSInterfaceDeclaration` we need to read the actual location from the comment node and use that to find out where in the `TSInterfaceDeclaration` the comment was.